### PR TITLE
add snackbar with network errors

### PR DESCRIPTION
**Pull request by toddmedema**
screen shot 2017-05-18 at 11 54 55 pm

Closes https://github.com/ExpeditionRPG/expedition-app/issues/142

### CODE REVIEW

File: app/reducers/Snackbar.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+};
+
+export function snackbar(state: SnackbarState = default_state, action: Redux.Action): SnackbarState {
+  switch(action.type) {
+    case 'SNACKBAR_OPEN':
+      const openAction = (action as SnackbarOpenAction);
+      return {
+        open: true,
+        message: openAction.message,
+        timeout: openAction.timeout || 4000,
```
Comment (smartin015) [inline app/reducers/Snackbar.tsx]: Does this mean `state.snackbar.open` will be open even after the snackbar has timed out?
If so, thoughts on just storing the message and a timestamp at which the snackbar closes?

File: app/actions/Web.tsx [hunk 2]
```diff
@@ -59,10 +62,10 @@ export function search(numPlayers: number, user: UserState, search: SearchSettin
     // TODO: Pagination
     xhr.open('POST', authSettings.urlBase + '/quests', true);
     xhr.setRequestHeader('Content-Type', 'text/plain');
-    xhr.onload = function() {
+    xhr.onload = () => {
       var response: any = JSON.parse(xhr.responseText);
       if (response.error) {
-        throw new Error(response.error);
+        return dispatch(openSnackbar('Network error when searching: ' + response.error));
```
Comment (smartin015) [inline app/actions/Web.tsx]: What happens when `response.error` is enormous?

File: app/actions/Web.tsx [hunk 3]
```diff
@@ -93,7 +99,7 @@ export function subscribe(email: string) {
       if (xhr.status === 200) {
         window.FirebasePlugin.logEvent('user_subscribe', email);
       } else {
-        console.log('Error encountered when subscribing: ' + err);
+        dispatch(openSnackbar('Error when subscribing: ' + err));
```
Comment (smartin015) [inline app/actions/Web.tsx]: Same question here.

File: app/actions/Web.tsx [hunk 4]
```diff
@@ -128,10 +134,10 @@ export function submitUserFeedback(quest: QuestState, settings: SettingsType, us
       if (xhr.status === 200) {
         window.FirebasePlugin.logEvent('user_feedback_' + userFeedback.type, data);
         dispatch(userFeedbackClear());
-        return alert('Submission successful. Thank you!');
+        return dispatch(openSnackbar('Submission successful. Thank you!'));
       }
       window.FirebasePlugin.logEvent('user_feedback_' + userFeedback.type + '_err', data);
-      alert('Error encountered when submitting: ' + err);
+      return dispatch(openSnackbar('Error submitting feedback: ' + err));
```
Comment (smartin015) [inline app/actions/Web.tsx]: And here.

File: app/reducers/Snackbar.tsx (hunk 1 above)
Comment (toddmedema) [inline app/reducers/Snackbar.tsx]: Nope - the snackbar material UI API fires the snackbar close event on timeout, which we then have set open to false

File: app/actions/Web.tsx (hunk 2 above)
Comment (toddmedema) [inline app/actions/Web.tsx]: Just tested - truncates it to a single line

**Pull request merged**